xarray/tests/test_backends.py

```diff
@@ -1955,6 +1955,39 @@ def test_dump_encodings_h5py(self):
             assert actual.x.encoding['compression_opts'] is None
 
 
+@requires_h5fileobj
+class TestH5NetCDFFileObject(TestH5NetCDFData):
+    h5py = pytest.importorskip('h5py', minversion='2.9.0')
+    engine = 'h5netcdf'
+
+    @network
+    def test_h5remote(self):
+        # alternative: http://era5-pds.s3.amazonaws.com/2008/01/main.nc
+        import requests
+        url = ('https://www.unidata.ucar.edu/'
+               'software/netcdf/examples/test_hgroups.nc')
```

Comment:
Rather than going over the network, it might be quite easy to fire up a [`http.server.SimpleHTTPRequestHandler`](https://docs.python.org/3/library/http.server.html#http.server.SimpleHTTPRequestHandler) as part of a fixture. This would allow us to test the remote capability without internet (and without depending on a third party to host a file.)

Reply:
thanks for the suggestion @rabernat, i've gone with shoyers suggestion below to scrap the network test for now